quinn/src/connection.rs

```diff
@@ -1069,3 +1073,10 @@ pub enum SendDatagramError {
     #[error("connection closed: {0}")]
     ConnectionClosed(#[source] ConnectionError),
 }
+
+/// The maximum amount of datagrams that are sent in a single transmit
+///
+/// This can be lower than the maximum platform capabilities, to avoid excessive
+/// memory allocations when calling `poll_transmit()`. Benchmarks have shown
+/// that numbers around 10 are a good compromise.
+const MAX_TRANSMIT_SEGMENTS: usize = 10;
```

Comment:
Similarly, should we cap this in the quinn-proto layer instead of in the quinn layer?

Reply:
I don't think it matters a lot, and I'm fine either way. This might be rather temporary anyway. Follow-up changes should either
- look into using bigger pooled buffers to reduce the cost of transmitting more segments
- passing a buffer to `poll_transmit`, which would already provide a hint about the maximum size upfront

Reply:
If we're capping for the benefit of limiting memory allocations in quinn-proto, we should also cap it there IMO.

Reply:
Moving the cap to -proto seems reasonable to me.

Reply:
I moved the cap